src/server/ws_connection.py

```diff
@@ -376,7 +384,7 @@ async def _read_one_message(self) -> Optional[Message]:
             self.bytes_read += len(data)
             self.last_message_time = time.time()
             if not self.inbound_rate_limiter.process_msg_and_check(full_message_loaded):
-                if self.local_type == NodeType.FULL_NODE:
+                if self.local_type == NodeType.FULL_NODE and not is_localhost(self.peer_host):
```

Comment:
I think we should drop a WARNING if we would have rate limited ourselves as I think it hints at a bug

Reply:
There is a warning in line 388